pkg/cmd/cli/cmd/env.go

```diff
@@ -234,7 +243,14 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	}
 
 	skipped := 0
-	for _, info := range infos {
+	oldData := make([][]byte, len(infos))
+	for i, info := range infos {
+		old, err := json.Marshal(info.Object)
```

Comment:
I guess Latest() populates VersionedObject from the API, which (maybe?) forces it to the API version the client will use? Can you test this with input from a file containing a `v1beta3` object to make sure `v1` is used to create the patch and submit to the server?